tensornetwork/linalg/linalg.py

```diff
@@ -0,0 +1,191 @@
+                             dtype=dtype, M=M)
+  return the_node
+
+
+def zeros(shape: Sequence[int],
+          dtype: Optional[Type[np.number]],
+          name: Optional[Text] = None,
+          axis_names: Optional[List[Text]] = None,
+          backend: Optional[Union[Text, BaseBackend]] = None) -> Tensor:
+  """Return a Node of shape 'shape' of all zeros.
+  The Node has one dangling Edge per dimension.
+  Args:
+    shape : Shape of the array.
+    dtype, optional: dtype of array.
+    name (text, optional): Name of the Node.
+    axis_names (optional): List of names of the edges.
+    backend (optional): The backend or its name.
+  Returns:
+    the_node : Node of shape 'shape'
+        Represents an array of all zeros.
+  """
+  the_node = initialize_node("zeros", shape,
+                             name=name, axis_names=axis_names, backend=backend,
+                             dtype=dtype)
+  return the_node
+
+
+def ones(shape: Sequence[int],
+         dtype: Optional[Type[np.number]],
+         name: Optional[Text] = None,
+         axis_names: Optional[List[Text]] = None,
+         backend: Optional[Union[Text, BaseBackend]] = None) -> Tensor:
+  """Return a Node of shape 'shape' of all ones.
+  The Node has one dangling Edge per dimension.
+  Args:
+    shape : Shape of the array.
+    dtype, optional: dtype of array.
+    name (text, optional): Name of the Node.
+    axis_names (optional): List of names of the edges.
+    backend (optional): The backend or its name.
+  Returns:
+    the_node : Node of shape 'shape'
+        Represents an array of all ones.
+  """
+  the_node = initialize_node("ones", shape,
+                             name=name, axis_names=axis_names, backend=backend,
+                             dtype=dtype)
+  return the_node
+
+
+def randn(shape: Sequence[int],
+          dtype: Optional[Type[np.number]],
+          seed: Optional[int] = None,
+          name: Optional[Text] = None,
+          axis_names: Optional[List[Text]] = None,
+          backend: Optional[Union[Text, BaseBackend]] = None) -> Tensor:
+  """Return a Node of shape 'shape' of Gaussian random floats.
+  The Node has one dangling Edge per dimension.
+  Args:
+    shape : Shape of the array.
+    dtype, optional: dtype of array.
+    seed, optional: Seed for the RNG.
+    name (text, optional): Name of the Node.
+    axis_names (optional): List of names of the edges.
+    backend (optional): The backend or its name.
+  Returns:
+    the_node : Node of shape 'shape'
+        Represents an array of all ones.
+  """
+  the_node = initialize_node("randn", shape,
+                             name=name, axis_names=axis_names, backend=backend,
+                             seed=seed, dtype=dtype)
+  return the_node
+
+
+def random_uniform(shape: Sequence[int],
+                   dtype: Optional[Type[np.number]],
+                   seed: Optional[int] = None,
+                   boundaries: Optional[Tuple[float, float]] = (0.0, 1.0),
```

Comment:
apparently `Optional` means that it can be `None` as well. If that is not possible then  remove `Optional`